web/templates/settings/edit_custom_profile_field_form.hbs

```diff
@@ -1,53 +1,55 @@
 {{#with profile_field_info}}
-<form class="name-setting profile-field-form new-style" id="edit-custom-profile-field-form-{{id}}">
+<form class="name-setting profile-field-form new-style" id="edit-custom-profile-field-form-{{id}}" data-profile-field-id="{{id}}">
     <div class="input-group">
         <label for="name">{{t "Label" }}</label>
-        <input type="text" name="name" class="modal_text_input" value="{{ name }}" maxlength="40" />
+        <input type="text" name="name" id="id-custom-profile-field-name" class="modal_text_input prop-element" value="{{ name }}" maxlength="40" data-setting-widget-type="string" />
     </div>
     <div class="input-group hint_change_container">
         <label for="hint">{{t "Hint" }}</label>
-        <input type="text" name="hint" class="modal_text_input" value="{{ hint }}" maxlength="80" />
+        <input type="text" name="hint" id="id-custom-profile-field-hint" class="modal_text_input prop-element" value="{{ hint }}" maxlength="80" data-setting-widget-type="string" />
     </div>
-    {{#if is_select_field }}
-    <div class="input-group">
-        <label for="profile_field_choices_edit">{{t "Field choices" }}</label>
-        <div class="profile-field-choices" name="profile_field_choices_edit">
-            <hr />
-            <div class="edit_profile_field_choices_container">
-                {{#each choices}}
-                    {{> profile_field_choice }}
-                {{/each}}
+    <span class="prop-element" id="id-custom-profile-field-field-data" data-setting-widget-type="field-data-setting">
```

Comment:
The direct parent of both field choices and external account input is the form, don't think we should add the attributes there since it might seem like the whole form is `field_data`. If you were talking about putting the attributes in either one of the profile field choices or external account div, e.g. If you add them to the div for profile field choices. The div will not be part of the dom when `is_external_account_field` is true, and thus it will not get picked up by `get_subsection_property_elements`.

Reply:
Ah, I see what you mean now, that will definitely work, the`id` will still remain globally unique since only one of the condition will be true. 

That technically works for both, but the html from `is_external_account_field` has 2 divs inside it. I can put the attributes on any one of them and it will work since we're just fetching the parent right away in `get_field_data_input_value`, but it might confuse someone who is just looking at the handlebars file directly, they might assume that the div without the attributes does not contain `field_data` when it actually does. We can avoid this readability problem by adding an element inside the `is_external_account_field` if block, but that's what we were trying to avoid in the first place.

LMK what you think @sahil839

Reply:
I think it is fine to add the extra element inside `is_external_account_field` if block considering that the we want to check whether the `field_data` value is changed and that is the combination of two inputs.